backend/compact-connect/lambdas/python/data-events/tests/function/test_encumbrance_events.py

```diff
@@ -3031,3 +3031,250 @@ def test_license_encumbrance_lifting_notification_listener_skips_notifications_w
         # Verify NO notifications were sent because license is still encumbered
         mock_provider_email.assert_not_called()
         mock_state_email.assert_not_called()
+
+    @patch('cc_common.event_bus_client.EventBusClient._publish_event')
+    def test_license_encumbrance_listener_does_not_create_duplicate_update_records_for_unencumbered_privileges_on_retry(
+        self,
+        _mock_publish_event,
+    ):
+        """Test that license encumbrance event does not create duplicate update records
+        when re-run for unencumbered privileges."""
+        from cc_common.data_model.schema.common import UpdateCategory
+        from handlers.encumbrance_events import license_encumbrance_listener
+
+        # Set up test data
+        self.test_data_generator.put_default_provider_record_in_provider_table()
+        privilege = self.test_data_generator.put_default_privilege_record_in_provider_table(
+            value_overrides={
+                'licenseJurisdiction': DEFAULT_LICENSE_JURISDICTION,
+                'licenseTypeAbbreviation': DEFAULT_LICENSE_TYPE_ABBREVIATION,
+                'encumberedStatus': 'unencumbered',
+                'jurisdiction': 'ne',
+            }
+        )
+
+        # Add adverse action for license
+        self.test_data_generator.put_default_adverse_action_record_in_provider_table(
+            value_overrides={'actionAgainst': 'license'}
+        )
+
+        message = self._generate_license_encumbrance_message()
+        event = self._create_sqs_event(message)
+
+        # Execute the handler FIRST TIME
+        license_encumbrance_listener(event, self.mock_context)
+
+        # Verify privilege update record was created
+        serialized_privilege = privilege.serialize_to_database_record()
+        privilege_update_records = self._provider_table.query(
```

Comment:
We're also trying to move away from 'peeking under the hood' in tests, where possible (as in querying directly with hard-coded pk/sk), in favor of using established interfaces that abstract the handling of particular queries. Here's an example:
https://github.com/InspiringApps/JCC-CompactConnect/blob/ff8c7c05c240d531f69c4502268337f607df5ff1/backend/compact-connect/lambdas/python/data-events/tests/function/test_encumbrance_events.py#L247